perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -5097,7 +5097,8 @@ private UserExtSourceAttributesModuleImplApi getUserExtSourceAttributeModule(Per
 	 */
 	GroupResourceVirtualAttributesModuleImplApi getResourceGroupVirtualAttributeModule(PerunSession sess, AttributeDefinition attribute) throws InternalErrorException {
 		Object attributeModule = getAttributesModule(sess, attribute);
-		if (attributeModule == null) return null;
+		if (attributeModule == null)
+			throw new ModuleNotExistsException("Virtual attribute module for " + attribute + " doesn't exists.");
```

Comment:
I just would like to ask, why is this change made only to these three methods and not to all of them? This way, the module accessing methods behave differently for different combinations of entities.

Reply:
Module is "required" only for virtual attributes, since it implements ``getAttributeValue()`` to actually compute the value. Other getters for virtual attribute modules were already correct, so this commit fixes implementation to be consistent for all virtual attributes.

Reply:
I see, thanks for the explanation.